x-pack/filebeat/input/awss3/input.go

```diff
@@ -376,5 +380,24 @@ func getProviderFromDomain(endpoint string, ProviderOverride string) string {
 	return "unknown"
 }
 
+func pollSqsWaitingMetric(ctx context.Context, receiver *sqsReader) {
+	t := time.NewTicker(time.Minute)
+	defer t.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-t.C:
+			count, err := receiver.GetApproximateMessageCount(ctx)
+			if count == -1 && strings.Contains(err.Error(), "StatusCode: 403") {
```

Comment:
How did you determine that 403 is the status code? Based on the [common error codes](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/CommonErrors.html) I would have expected a AccessDeniedException with a 400, but I'm not 100% sure.

Checking error strings is usually only done as a last resort because it is brittle. It's easy for the formatting to change and then this code no longer works. Here's an example of checking the error code. https://github.com/elastic/beats/blob/a461c17597c474d6a69bebcaf405fe581a5aecda/x-pack/filebeat/input/awss3/sqs_s3_event.go#L203-L212

You could probably apply that pattern here.

Reply:
I thought I managed to mock this scenario for live testing, and that's where I got the 403. But after digging in deeper, you're right and I should be getting the `AccessDeniedException` / 400. 

I updated to use this pattern